File: content/code-examples/tested/javascript/driver/aggregation/pipelines/join-multi-field/tutorial.snippet.embedded-pl-match-name-variation.js
const embedded_pl = [];

embedded_pl.push({
  $match: {
    $expr: {
      $and: [
        { $eq: ['$product_name', '$$prdname'] },
        { $eq: ['$product_variation', '$$prdvartn'] },
      ],
    },
  },
});

File: content/code-examples/tested/java/driver-sync/aggregation/pipelines/joinMultiField/Tutorial.snippet.embedded-pl-match-name-variation.java
List<Bson> embeddedPipeline = new ArrayList<>();

embeddedPipeline.add(Aggregates.match(
        Filters.expr(
                Filters.and(
                        new Document("$eq", Arrays.asList("$product_name", "$$prdname")),
                        new Document("$eq", Arrays.asList("$product_variation", "$$prdvartn"))
                )
        )
));
